gnuradio-runtime/lib/hier_block2_detail.cc

```diff
@@ -496,9 +496,27 @@ endpoint_vector_t hier_block2_detail::resolve_port(int port, bool is_input)
     return result;
 }
 
+void hier_block2_detail::recursive_disconnect_all(const std::string& caller)
+{
+    d_debug_logger->debug("Disconnect hier_block2 recursive...");
+    std::vector<basic_block_vector_t> paths = d_fg->partition();
+    for (std::vector<basic_block_vector_t>::iterator p = paths.begin(); p != paths.end(); p++) {
+        for (basic_block_viter_t bp = (*p).begin(); bp != (*p).end(); bp++) {
+            if (this->d_owner->identifier() != caller) {
```

Comment:
Does `this->d_owner` change in these loops? If not, then this should be an outer condition, not an inner condition: "If we are already owned by the caller, don't recursively descend" is better than "recursively descend and while we're doing so, check for every node we find whether we might be owned by the caller"